woo-razorpay.php

```diff
@@ -1824,6 +1849,140 @@ public function update1ccOrderWC(& $order, $wcOrderId, $razorpayPaymentId)
+                    $giftCode = $giftcard['code'];
+                    if(is_plugin_active('yith-woocommerce-gift-cards/init.php')){
+
+                        $yithCard = new YITH_YWGC_Gift_Card( $args = array('gift_card_number'=> $giftCode));
+
+                        $giftCardBalance = $yithCard->get_balance();
+
+                        if($giftCardBalance == null && $balance >= 0 && $usedAmt > $giftCardBalance ){
+                            // initiate refund in case gift card faliure
+                            $this->process_refund($orderId, $razorpayData['amount_paid'], $reason = '', $razorpayPaymentId);
+                        }else{
+                            //Deduct amount of gift card
+                            $yithCard->update_balance( $yithCard->get_balance() - $usedAmt );
+                            $yithCard->register_order($orderId);
+
+                            $code = "Gift Card ($giftCode)";
+
+                            $wpdb->query( // phpcs:ignore
+                                $wpdb->prepare(
+                                    'INSERT INTO `' . $wpdb->prefix . 'woocommerce_order_items`( order_item_name, order_item_type, order_id ) VALUES ( %s, %s, %s)',
+                                    $code,
+                                    'fee',
+                                    $orderId
+                                )
+                            );
+
+                            $itemId = $wpdb->insert_id;
+                            wc_add_order_item_meta( $itemId, '_line_total', -$usedAmt );
+                        }
+
+                    }
+                    else if(is_plugin_active('pw-woocommerce-gift-cards/pw-gift-cards.php'))
+                    {
+                        $result = $wpdb->get_row( $wpdb->prepare( "SELECT * FROM `{$wpdb->pimwick_gift_card}` WHERE `number` = %s", $giftcard['code'] ) );
+                        if($result != null){
+                            $balance = $wpdb->get_var( $wpdb->prepare( "SELECT SUM(amount) FROM {$wpdb->pimwick_gift_card_activity} WHERE pimwick_gift_card_id = %d", $result->pimwick_gift_card_id ) );
+                            
+                            if($balance == null && $balance >= 0 && $usedAmt > $balance ){
+                                // initiate refund in case gift card faliure
+                                $this->process_refund($orderId, $razorpayData['amount_paid'], $reason = '', $razorpayPaymentId);
+                            }else{
+                                //Deduct amount of gift card
+                               $this->debitGiftCards($orderId, $order, "order_id: $orderId checkout_update_order_meta", $usedAmt);
+                            }
+
+                        }else{
+                            $this->process_refund($orderId, $razorpayData['amount_paid'], $reason = '', $razorpayPaymentId);
+                        }
+
+                    }else{
```

Comment:
If GC is applied then the customer can't apply coupons?

Reply:
Customer can use both at a time